### DISCUSSION TITLE
I've followed all tutorials and none have worked to upload Unity Project

### DISCUSSION THREAD

**Discussion by KevinHJoe** [community/community#23784] [New to GitHub]
<p>I can’t figure out how to use GitHub for the life of me. Is it supposed to be incredibly hard on purpose?<br>
As the title has suggested, i’ve watched all the tutorials on youtube on how to upload a Unity project to GitHub, but I still can’t actually upload my project to my Repository.<br>
There must be some secret that all these tutorials are not including.<br>
What could I possibly be doing wrong?</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/travelsbyfire">@travelsbyfire</a>!</p>
<p>We’re going to need some more information to be able to help you out. Could you outline the steps you’ve taken, what you expect to be happening, and what is happening instead? That way, we can figure out where you’re stuck.</p>
<p>Thanks!</p>

**Comment by KevinHJoe** [comment]
<p>So what I’ve tried was  using the GitHub Desktop version. I first create a repo on GitHub, then link that repo to the GitHub desktop. I create a folder for the .gitignore file, and then I put my Unity project inside that folder. I’ve been told that the .gitignore file goes at the root of the folder, and I’m pretty sure that’s how I have it, where it’s outside the Unity project folder, but inside a folder that’s holding both of them.<br>
Something that I haven’t figured out is how to Commit to Master. Up until this point, all the tutorials do exactly what I’ve described, but then they are able to hit the Commit to Master button and essentially upload their projects. However, I can’t hit the Commit to Master button as it’s greyed out. I’ve tried pushing the files, adding other commits, but nothing seems to make the Commit to Master button work.</p>

**Comment by steveward** [comment, marked as answer]
<p><a class="mention" href="/travelsbyfire">@travelsbyfire</a> thanks for sharing those details.  The <code>Commit to master</code> button will stay disabled until you enter a commit message. Can you verify that you have entered a commit message in the <code>Summary (required)</code> field below the list of your changes?</p>
<p>If that does not get things working it would be helpful if you could share a screenshot of what you are seeing – that may give us a better idea of what the issue could be.</p>

**Comment by KevinHJoe** [comment]
<p>Ok well now I feel silly. I don’t recall any of the tutorials doing this, but just adding a title to the commit allows me to use the Commit to Master button. After that I pushed the files, and am able to see the files on GitHub… Thank you</p>
